lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -0,0 +1,554 @@
+
+                this.updateValueInField();
+
+                if (this.editable === 'toggle') {
+                    this.showCanEditMessage(this.mode === 'display');
+                }
+                return this;
+            },
+
+            modelValueIsSet: function() {
+                var value = this.modelValue();
+                if (_.isUndefined(value) || _.isNull(value) || value == '') {
+                    return false;
+                } else {
+                    return !(_.isUndefined(this.optionForValue(value)))
+                }
+            },
+
+            optionForValue: function(value) {
+                return _.find(this.options.options, function(option) { return option[0] == value; })
+            },
+
+            fieldValue: function () {
+                return this.$('.u-field-value select').val();
+            },
+
+            displayValue: function (value) {
+                if (value) {
+                    var option = this.optionForValue(value);
+                    return (option ? option[1] : '');
+                } else {
+                    return '';
+                }
+            },
+
+            updateValueInField: function () {
+                var value = this.modelValue();
+                if (this.mode === 'display') {
+                    if (_.isUndefined(value) || _.isNull(value)) {
+                        value = this.options.placeholderValue || '';
+                    }
+                    this.$('.u-field-value').html(Mustache.escapeHtml(this.displayValue(value)));
+                    this.showDisplayMode(false);
+                } else {
+                    if (_.isUndefined(value) || _.isNull(value)) {
+                        value = '';
+                    }
+                    this.$('.u-field-value select').val(Mustache.escapeHtml(value));
+                }
+            },
+
+            saveValue: function () {
+                var attributes = {};
+                attributes[this.options.valueAttribute] = this.fieldValue();
+                this.saveAttributes(attributes);
+            },
+
+            showEditMode: function(render) {
+                this._super(render);
+            },
+
+            showSuccessMessage: function() {
+                this._super();
+                if (this.editable === 'toggle') {
+                    this.showDisplayMode(true);
+                }
+            }
+        });
+
+        AccountSettingsFieldViews.AccountPrivacyFieldView = AccountSettingsFieldViews.DropdownFieldView.extend({
+
+            render: function () {
+                this._super();
+                this.message();
+                return this;
+            },
+
+            message: function (message) {
+                if (this.profileIsPrivate) {
+                    this._super(interpolate_text(
+                        gettext('Your profile is set to limited because you have not set your birth year yet. Set it on the {account_settings_page_link}.'),
```

Comment:
Text: Before you share your full profile, you must set your birth year. To set your birth year, go to the {account_settings_page_link}.